Tests/SonarScanner.MSBuild.PreProcessor.Tests/PreProcessorTests.cs

```diff
@@ -267,6 +261,127 @@ public void PreProc_EndToEnd_SuccessCase_With_Organization()
             AssertAnalysisConfig(settings.AnalysisConfigFilePath, 2, logger);
         }
 
+        [DataTestMethod]
+        [DataRow("6.7.0.22152")]
+        [DataRow("7.0.0.22152")]
+        public void PreProc_EndToEnd_ShouldWarn_SonarQubeDeprecatedVersion(string sqVersion)
```

Comment:
It was expected, but it took a significant amount of time to verify that these two long tests are copy/pasted, except the last line. Can you extract the main part into a private method returning `mockServer` to assert?

Reply:
Other way could be using `DataRow`, let's say for 6.7, SC and 10.0 with `bool shouldLogWarning` parameter. And do something like this:
```
if(shouldLogWarning)
  mockServer.AssertWarningWritten("version is below supported");
else
  mockServer.AssertNoWarningWritten();
```